### PR TITLE
[MRG+1] Update hyper-parameter search example to use decorator

### PR DESCRIPTION
**Pull request by betatim**
Make use of the new decorator for keyword arguments to objectives.

### CODE REVIEW

File: examples/hyperparameter-optimization.ipynb [hunk 1]
```diff
@@ -82,81 +66,70 @@
...
+    "\n",
+    "\n",
+    "# The list of hyper-parameters we want to optimize. For each one we define the bounds,\n",
+    "# the corresponding scikit-learn parameter name, as well as how to sample values\n",
+    "# from that dimension (`'log-uniform'` for the learning rate)\n",
+    "space  = [sk_space.Integer(1, 5, name='max_depth'),\n",
+    "          sk_space.Real(10**-5, 10**0, \"log-uniform\", name='learning_rate'),\n",
+    "          sk_space.Integer(1, n_features, name='max_features'),\n",
+    "          sk_space.Integer(2, 100, name='min_samples_split'),\n",
+    "          sk_space.Integer(1, 100, name='min_samples_leaf')]\n",
```
Comment (glouppe) [inline examples/hyperparameter-optimization.ipynb]: `from skopt.space import Real, Integer`? that would remove some of the boilerplate.

Comment (betatim) [reply examples/hyperparameter-optimization.ipynb]: 👍

**Comment by glouppe** [comment]
LGTM!

**Comment by betatim** [comment]
Merged! Get those travis runs done before the US wakes up :)

**Pull request merged**